runtime/testy/Ansible.py

```diff
@@ -0,0 +1,176 @@
+
+    def threadMaker(self, threadTarget, threadName):
+        thread = threading.Thread(target = threadTarget,
+                                  name = threadName,
+                                  args = (self, self.badThingsQueue, self.stateQueue, self.pipe))
+        thread.daemon = True
+        return thread
+
+    def start(self):
+        packagerThread = self.threadMaker(self.packagerFunc, self.packagerName)
+        socketThread = self.threadMaker(self.socketFunc, self.socketName)
+        packagerThread.start()
+        socketThread.start()
+        while True:
+            time.sleep(1)
+
+
+
+class UDPSendClass(AnsibleHandler):
+    SEND_PORT = 1235
+
+    def __init__(self, badThingsQueue, stateQueue, pipe):
+        self.sendBuffer = TwoBuffer()
+        packagerName = THREAD_NAMES.UDP_PACKAGER
+        sockSendName = THREAD_NAMES.UDP_SENDER
+        super().__init__(packagerName, sockSendName, UDPSendClass.packageData,
+                         UDPSendClass.udpSender, badThingsQueue, stateQueue, pipe)
+
+    def packageData(self, badThingsQueue, stateQueue, pipe):
+        """Function run as a thread that packages data to be sent.
+
+        The robot's current state is received from the StateManager via the pipe and packaged 
+        by the package function, defined internally. The packaged data is then placed 
+        back into the TwoBuffer replacing the previous state.
+        """
+        def package(state, badThingsQueue):
+            """Helper function that packages the current state.
+
+            Currently this function converts the input into bytes. This will
+            eventually be implemented to package the rawState into protos.
+            """
+            s = "TEST" 
+            b = bytearray()
+            b.extend(map(ord, s))
+            return b 
+
+        while True:
+            try:
+                stateQueue.put([SM_COMMANDS.SEND_ANSIBLE, []])
+                rawState = pipe.recv()
+                if rawState == RUNTIME_CONFIG.PIPE_READY:
+                    stateQueue.put([SM_COMMANDS.SEND_ANSIBLE, []])
+                elif rawState:
+                    packState = package(rawState, badThingsQueue)
+                    self.sendBuffer.replace(packState) 
+            except Exception:
+                badThingsQueue.put(BadThing(sys.exc_info(), 
+                    "UDP packager thread has crashed with error:",  
+                    event = BAD_EVENTS.UDP_SEND_ERROR, 
+                    printStackTrace = True))
+
+    def udpSender(self, badThingsQueue, stateQueue, pipe):
+        """Function run as a thread that sends a packaged state from the TwoBuffer
+        
+        The current state that has already been packaged is gotten from the 
+        TwoBuffer, and is sent to Dawn via a UDP socket.
+        """
+        host = socket.gethostname() #TODO: determine host in runtime-dawn comm
+        with socket.socket(socket.AF_INET, socket.SOCK_DGRAM) as s:
+            while True: 
```

Comment:
This loop needs a time.sleep in it somewhere. Currently, this is just constantly sending without any rate limiting. 

time.sleep at some set interval configurable in runtime.util would be my recommendation.